homeassistant/components/deconz/sensor.py

```diff
@@ -52,35 +53,55 @@
 ATTR_DAYLIGHT = "daylight"
 ATTR_EVENT_ID = "event_id"
 
-DEVICE_CLASS = {
-    Consumption: DEVICE_CLASS_ENERGY,
-    Humidity: DEVICE_CLASS_HUMIDITY,
-    LightLevel: DEVICE_CLASS_ILLUMINANCE,
-    Power: DEVICE_CLASS_POWER,
-    Pressure: DEVICE_CLASS_PRESSURE,
-    Temperature: DEVICE_CLASS_TEMPERATURE,
-}
-
-ICON = {
-    Daylight: "mdi:white-balance-sunny",
-    Pressure: "mdi:gauge",
-    Temperature: "mdi:thermometer",
-}
-
-STATE_CLASS = {
-    Consumption: STATE_CLASS_TOTAL_INCREASING,
-    Humidity: STATE_CLASS_MEASUREMENT,
-    Pressure: STATE_CLASS_MEASUREMENT,
-    Temperature: STATE_CLASS_MEASUREMENT,
-}
-
-UNIT_OF_MEASUREMENT = {
-    Consumption: ENERGY_KILO_WATT_HOUR,
-    Humidity: PERCENTAGE,
-    LightLevel: LIGHT_LUX,
-    Power: POWER_WATT,
-    Pressure: PRESSURE_HPA,
-    Temperature: TEMP_CELSIUS,
+ENTITY_DESCRIPTIONS = {
+    Battery: SensorEntityDescription(
+        key="battery",
+        device_class=DEVICE_CLASS_BATTERY,
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=PERCENTAGE,
+    ),
+    Consumption: SensorEntityDescription(
+        key="consumption",
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    Daylight: SensorEntityDescription(
+        key="daylight",
+        icon="mdi:white-balance-sunny",
+        entity_registry_enabled_default=False,
+    ),
+    Humidity: SensorEntityDescription(
+        key="humidity",
+        device_class=DEVICE_CLASS_HUMIDITY,
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=PERCENTAGE,
+    ),
+    LightLevel: SensorEntityDescription(
+        key="lightlevel",
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        unit_of_measurement=LIGHT_LUX,
+    ),
+    Power: SensorEntityDescription(
+        key="power",
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=POWER_WATT,
+    ),
+    Pressure: SensorEntityDescription(
+        key="pressure",
+        device_class=DEVICE_CLASS_PRESSURE,
+        icon="mdi:gauge",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=PRESSURE_HPA,
+    ),
+    Temperature: SensorEntityDescription(
+        key="temperature",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        icon="mdi:thermometer",
```

Comment:
If we set a device class we normally don't need to set a custom icon since the frontend sets a specific icon for the device class already.

Reply:
Great. I'll clean that up

Reply:
Done